pkg/daemon/ceph/client/filesystem.go

```diff
@@ -134,16 +144,79 @@ func CreateFilesystem(context *clusterd.Context, clusterName, name, metadataPool
+func IsMultiFSEnabled() bool {
+	t := os.Getenv(MultiFsEnv)
+	if t == "true" {
+		return true
+	}
+	return false
+}
+
+// SetNumMDSRanks sets the number of mds ranks (max_mds) for a Ceph filesystem.
+func SetNumMDSRanks(context *clusterd.Context, clusterName, fsName string, activeMDSCount int32) error {
+	// Noted sections 1 and 2 are necessary for reducing max_mds in Luminous.
+	//   See more:   [1] http://docs.ceph.com/docs/luminous/cephfs/upgrading/
+	//               [2] https://tracker.ceph.com/issues/23172
+	// TODO: These two sections could be conditionally skipped if this function can be informed that
+	//   the version of Ceph mdses currently running is >=13.0.0
+
+	// * Noted section 1 - See note at top of function
+	fsAtStart, errAtStart := GetFilesystem(context, clusterName, fsName)
+	// collect information now, but don't check error yet
+	// * End of Noted section 1
+
+	// Always tell Ceph to set the new max_mds value
+	args := []string{"fs", "set", fsName, "max_mds", strconv.Itoa(int(activeMDSCount))}
+	if _, err := ExecuteCephCommand(context, clusterName, args); err != nil {
+		return fmt.Errorf("failed to set filesystem %s num mds ranks (max_mds) to %d: %v",
+			fsName, activeMDSCount, err)
+	}
+
+	// ** Noted section 2 - See note at top of function
+	// Now check the error to see if we can even determine whether we should reduce or not
+	if errAtStart != nil {
+		return fmt.Errorf(`failed to get filesystem %s info needed to ensure mds rank can be changed correctly,
+if Ceph version is Luminous (12.y.z) and num active mdses (max_mds) was lowered, user must deactivate extra active mdses manually: %v`,
+			fsName, errAtStart)
+	}
+	if int(activeMDSCount) > fsAtStart.MDSMap.MaxMDS {
+		return nil // No need to deactivate mdses if we are raising max_mds
+	}
+	logger.Debugf("deactivating some running mdses for filesystem %s", fsName)
+	// Deactivate all mdses except desired number (N); arbitrarily choose first N to live
+	fs, err := GetFilesystem(context, clusterName, fsName)
+	if err != nil {
+		logger.Warningf(`Failed to get filesystem %s info needed to deactivate running mdses
+Using slightly stale info, this could (rarely) result in momentary loss of filesystem availability: %v`, fsName, err)
+		fs = fsAtStart // <-- Do the best we can to disable mdses that were active when we started
+		// Effects of stale info should be non-destructive, & unlikely the info is actually bad
+	}
+	if len(fs.MDSMap.In) > int(activeMDSCount) {
+		// Only try to deactivate mdses if there are more than desired.
+		for _, gid := range fs.MDSMap.In[activeMDSCount:] {
+			args := []string{"mds", "deactivate", fmt.Sprintf("%s:%d", fsName, gid)}
+			if _, err := ExecuteCephCommand(context, clusterName, args); err != nil {
+				logger.Warningf(`Possibly failed to deactivate filesystem %s mds w/ gid %d
+In mimic+ this command does nothing; in luminous this is non-ideal but not necessarily critical: %v`,
```

Comment:
Could you change that to `> Mimic` and have that logging message on one line instead of split to two.